packages/client/src/object/fetchSingle.ts

```diff
@@ -57,3 +58,31 @@ export async function fetchSingle<
 
   return result.data[0] as any;
 }
+
+export async function fetchSingleWithErrors<
+  Q extends ObjectOrInterfaceDefinition,
+  const A extends FetchPageArgs<Q, any, any>,
+>(
+  client: MinimalClient,
+  objectType: Q,
+  args: A,
+  objectSet: ObjectSet,
+): Promise<
+  Result<
+    Osdk<
+      Q,
+      A["includeRid"] extends true ? SelectArgToKeys<Q, A> | "$rid"
+        : SelectArgToKeys<Q, A>
+    >
+  >
+> {
+  try {
+    const result = await fetchSingle(client, objectType, args, objectSet);
+    return { value: result as any };
+  } catch (e) {
+    if (e instanceof Error) {
+      return { error: e };
+    }
+    return { error: e as Error };
```

Comment:
Is this right?

Reply:
Should we have a wrapper function that can take a call and create a result based on if something throws?

Reply:
Hmm I'm confused, I was just emulating what we did for like `fetchPageWithErrors` https://github.com/palantir/osdk-ts/blob/d183d92b3fe0b2fa8016ea94d44eebc1e21fa2fc/packages/client/src/object/fetchPage.ts#L229

Reply:
Okay, lets go with it for now.